src/server/pps/server/api_server.go

```diff
@@ -2566,9 +2502,15 @@ func (a *apiServer) deletePipeline(ctx context.Context, request *pps.DeletePipel
 		}
 	}
 
-	// Delete PipelineInfo
+	// Delete all past PipelineInfos
 	if err := dbutil.WithTx(ctx, a.env.GetDBClient(), func(sqlTx *sqlx.Tx) error {
-		return a.pipelines.ReadWrite(sqlTx).Delete(request.Pipeline.Name)
+		for version := pipelineInfo.Version; version > 0; version-- {
```

Comment:
So pipelines start at version 1? Will there ever be gaps between existing pipeline version numbers?

Reply:
They start at 1, yes. This is definitely something that could be fragile, at the very least "not found" errors should be ignored here if they aren't already by the collection

Reply:
This now deletes based on an index